ecl/eclagent/eclagent.cpp

```diff
@@ -556,7 +556,8 @@ EclAgent::EclAgent(IConstWorkUnit *wu, const char *_wuid, bool _checkVersion, bo
     Owned<IWorkUnit> w = updateWorkUnit();
     if (_queryXML)
         w->setXmlParams(_queryXML);
-    updateSuppliedXmlParams(w);
+    if (w)
```

Comment:
Should this check be before w->setXmlParams(_queryXML); above as well?

Reply:
It wasn't protected before, but I will change since I think it could occur.